internal/controllers/cluster/cluster_controller_test.go

```diff
@@ -172,6 +172,50 @@ func TestClusterReconciler(t *testing.T) {
 		}, timeout).Should(BeTrue())
 	})
 
+	t.Run("Should successfully patch a cluster object if the spec diff is empty but the status conditions diff is not", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Setup
+		cluster := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test3-",
+				Namespace:    ns.Name,
+			},
+		}
+		g.Expect(env.Create(ctx, cluster)).To(Succeed())
+		key := client.ObjectKey{Name: cluster.Name, Namespace: cluster.Namespace}
+		defer func() {
+			err := env.Delete(ctx, cluster)
+			g.Expect(err).ToNot(HaveOccurred())
+		}()
+
+		// Wait for reconciliation to happen.
+		g.Eventually(func() bool {
+			if err := env.Get(ctx, key, cluster); err != nil {
+				return false
+			}
+			return len(cluster.Finalizers) > 0
+		}, timeout).Should(BeTrue())
+
+		// Patch
+		g.Eventually(func() bool {
+			ph, err := patch.NewHelper(cluster, env)
+			g.Expect(err).ToNot(HaveOccurred())
+			conditions.MarkTrue(cluster, clusterv1.InfrastructureReadyCondition)
```

Comment:
Is this a diff? Isn't the cluster controller also setting this condition to true in reconcile?

Reply:
Is the controller running? Wouldn't the above patch also be the same?

Reply:
l.192 seems to suggest that the controller is running :)

Reply:
Ran both tests through the debugger. The patch here is literally a no-op (same in the pre-existing test)